### PR TITLE
Fix View not found error classification

### PR DESCRIPTION
**Pull request by ajaygeorge**
## Description
View not found errors are currently classified as
GENERIC_INTERNAL_ERROR. This fix introduces a proper
error code VIEW_NOT_FOUND and classifies this as a USER_ERROR

## Motivation and Context
Correctly classify errors

## Test Plan
Unit Tests

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataHandle.java [hunk 1]
```diff
@@ -60,7 +62,10 @@ public void addTableColumnMetadata(QualifiedObjectName tableName, Future<TableCo
 
     public Optional<ViewDefinition> getViewDefinition(QualifiedObjectName viewName)
     {
-        checkState(viewDefinitions.containsKey(viewName), "View " + viewName + " not found, the available view names are " + viewDefinitions.keySet());
+        if (!viewDefinitions.containsKey(viewName)) {
+            throw new PrestoException(VIEW_NOT_FOUND, "View " + viewName + " not found, the available view names are " + viewDefinitions.keySet());
```
Comment (swapsmagic) [inline presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataHandle.java:66]: Let's follow the same pattern we use other places for the exception message as below:

`throw new PrestoException(VIEW_NOT_FOUND, "View %s not found, the available view names are: %s", viewName, viewDefinitions.keySet());`

**Review by swapsmagic — COMMENTED** [review]
Can we introduce a unit test to verify the exception as well?

**Comment by ajaygeorge** [comment]
@swapsmagic Added presto-analyzer/src/test/java/com/facebook/presto/sql/analyzer/MetadataHandleTest.java test case.

**Review by NikhilCollooru — APPROVED** [review]
LGTM

**Pull request merged**